Dynamic dashboard/UI tests

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #11453 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
This PR updates the UI tests to align with the dynamic dashboard changes.

Please feel free to share any np comments in the review, I don't have a lot of experience with UI tests especially ones that involve Compose, and if we can improve anything it would be great.

### Testing instructions
Just green CI.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Refresh visitors and revenue stats when date range changes

hichamboushaba committed:
Enable dynamic dashboard for tests

hichamboushaba committed:
Fix stats tests for dynamic dashboard

hichamboushaba committed:
Add a function that allows scrolling to specific Compose content

Specifically for Compose content contained in a scrollable Android View

hichamboushaba committed:
Update top performers UI tests for dynamic dashboard

hichamboushaba committed:
Improve readability

hichamboushaba committed:
Fix detekt issues

hichamboushaba committed:
Remove unused stubbing

hichamboushaba committed:
Make sure the selected site coroutine scope is canceled when tearing down tests

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Rename class to align more with the new naming

hichamboushaba committed:
Fix logout logic in UI tests

hichamboushaba committed:
Fix logic of canceling the site CoroutineScope

Resetting the selected site was causing failures due to the `SiteResetException`

hichamboushaba committed:
Restrict condition to check only for actual date range

0nko reviewed (approved):
LGTM 👍
